core/src/health.rs

```diff
@@ -0,0 +1,83 @@
+//! Module implementing shared basic health reporting.
+
+use crate::http_server::Handler;
+use anyhow::Result;
+use rouille::{Request, Response};
+use std::sync::atomic::{AtomicBool, Ordering};
+
+/// Trait for asyncronously notifying health information.
+#[cfg_attr(test, mockall::automock)]
+pub trait HealthReporting: Send + Sync {
+    /// Notify that the service is ready.
+    fn notify_ready(&self);
+}
```

Comment:
I don't think we need this trait because the only implementor is HttpHealthEndpoint and we don't need to mock it. Not a big deal to have it either or maybe you plan to have a second implementor.

Reply:
Its mostly to test that the driver only notifies that it is healthy after running the first batch.